services/121-service/src/payments/fsp-integration/safaricom/safaricom.api.service.ts

```diff
@@ -47,10 +78,33 @@ export class SafaricomApiService {
     } catch (error) {
       console.log(error, 'authenticate');
       console.error('Failed to make OAuth Access Token payment API call');
+
+      throw new SafaricomApiError(`Error: ${error.message}`);
     }
   }
 
-  public async transfer(payload: SafaricomTransferPayload): Promise<any> {
+  private createTransferPayload(
```

Comment:
Do not "blindly pass through" parameter objects from one function to the next. See guidelines.
In this case, since this is a private function and the class is pretty short, LOGAF is low.